feat(common): create from _helpers

devopsmariocom opened the pull request:
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/wc-catalogue/blaze-elements/blob/master/docs/CONTRIBUTING.md#commit
- [x] There are no linting errors and code is properly formatted (your run before push `yarn ts:format && yarn ts:lint:fix`)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

Closes #169 
- _helpers is simple folder
- packages are using relative paths for other packages

**What is the new behavior?**
- _helpers are now common package
- packages are linked using proper dependecies 

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

devopsmariocom committed:
feat(common): create from _helpers

Hotell commented:
readability nitpick, the title is "incomplete"

consider `feat(common): create regular package from _helpers`.

Also pls always provide links to existing issue, instead of writting by hand **What is the current behavior** section

devopsmariocom committed:
refactor(app-layout): use new component structure

devopsmariocom committed:
refactor(app-layout, calendar, card, drawer, modal, nav): fix type errors

affects: @blaze-elements/app-layout, blaze-elements, @blaze-elements/calendar, @blaze-elements/card, @blaze-elements/drawer, @blaze-elements/modal, @blaze-elements/nav, @blaze-elements/overlay

devopsmariocom committed:
style: fix style issues

devopsmariocom committed:
refactor(modal, button, selelect, overlay, card): fix minor issues/typos

devopsmariocom commented:
@Hotell tests are failing for unknown reason on CI, please meanwhile try go thru my changes.

Thanks

devopsmariocom committed:
chore(common): add ignored definition files

Hotell commented on packages/blaze-elements/index.demo.tsx:
```diff
@@ -8,6 +8,7 @@ import '@blaze-elements/button/index.demo';
 import '@blaze-elements/calendar/index.demo';
 import '@blaze-elements/card/index.demo';
 import '@blaze-elements/collapsible/index.demo';
+import '@blaze-elements/common/index.demo';
```
aha!

Hotell reviewed (commented):
pls do not merge. Tests are failing and there are some TS errors after our paired Code Review

Hotell commented:
depends on https://github.com/wc-catalogue/blaze-elements/issues/230

Hotell reviewed (changes requested):
depends on https://github.com/wc-catalogue/blaze-elements/issues/230. then merge with https://github.com/wc-catalogue/blaze-elements/pull/229

Hotell committed:
refactor: apply code review changes

* chore(common): add missing demo file

* refactor(drawer): move to new export package pattern

* refactor(calendar): use raw classes for imports across packages

* refactor(modal): use raw classes for imports across packages

* refactor(app-layout): use raw classes for imports across packages

* refactor(select): use raw classes for imports across packages

* fix(blaze-elements): add missing select to blaze-elements

* style: apply formatter

* fix(drawer): fix tsc errors to proper inferr to type definitions

* fix(select): fix wrong imports after refactor

* chore: remove tmp during cleanup

* test(nav): fix import after refactor

* test(blaze-elements): comment out select tests which are failing
